pinot-controller/src/main/java/org/apache/pinot/controller/api/resources/PinotRealtimeTableResource.java

```diff
@@ -77,14 +77,29 @@ public Response pauseConsumption(
   @POST
   @Path("/tables/{tableName}/resumeConsumption")
   @Produces(MediaType.APPLICATION_JSON)
-  @ApiOperation(value = "Resume consumption of a realtime table",
-      notes = "Resume the consumption for a realtime table")
+  @ApiOperation(value = "Resume consumption of a realtime table", notes =
+      "Resume the consumption for a realtime table. ConsumeFrom parameter indicates from which offsets "
+          + "consumption should resume. If consumeFrom parameter is not provided or 'best' option is chosen, "
+          + "consumption continues based on the offsets in segment ZK metadata, and in case the offsets are already "
+          + "gone, the first available offsets are picked to minimize the data loss.")
   public Response resumeConsumption(
-      @ApiParam(value = "Name of the table", required = true) @PathParam("tableName") String tableName) {
+      @ApiParam(value = "Name of the table", required = true) @PathParam("tableName") String tableName,
+      @ApiParam(value = "earliest | latest | best") @QueryParam("consumeFrom") String consumeFrom) {
     String tableNameWithType = TableNameBuilder.REALTIME.tableNameWithType(tableName);
     validate(tableNameWithType);
+    String offsetCriteria;
+    if ("earliest".equalsIgnoreCase(consumeFrom)) {
```

Comment:
why not just keep this as smallest/largest, since user is already familiar with those for table consumption?

Reply:
Sajjad originally had `offsetCriteria=smallest|largest`. I was the one asking him to change, and we are now at `consumeFrom=earliest|latest|best`.  

Since two of you prefer smallest/largest, I can go with that. I also felt that we should have a third (optional) value.  If we settle on `best`, then most people may choose that thinking it is some automatic way. In reality, what we mean is that we will pick up from where we left off.

If you both feel that leaving out the third option is ideal, I can go with that as well.

Let us wrap this up today, thanks.

Reply:
This is how I see pause/resume feature. Operator pauses consumption for some reason. After a while, they want to resume the consumption. The default behavior should be to pick up the events from the offset where we left off. If the offset is gone, we should automatically start with the smallest available offset. So that's the default behavior, but if operator wants to change the offset for some reason like stream connection change, then consumption should resume based on the provided "resumeFrom" parameter. As Neha mentioned, since users/operators are familiar with smallest/largest offset criteria, IMO it's better to use the same values.

Reply:
+1 on keeping smallest/largest. Imo no need of "best" as that's the default anyway.